src/azul/service/avro_pfb.py

```diff
@@ -153,24 +154,40 @@ def _add_missing_fields(cls, name: str, object_: MutableJSON, schema):
         None is the default value, but because of https://github.com/DataBiosphere/azul/issues/2370
         this isn't currently reflected in the schema.
         """
-        object_schema = one(f for f in schema['fields'] if f['name'] == 'object')
-        entity_schema = one(e for e in object_schema['type'] if e['name'] == name)
+        if schema['type'] == 'record':
+            object_schema = one(f for f in schema['fields'] if f['name'] == 'object')
+            entity_schema = one(e for e in object_schema['type'] if e['name'] == name)
+        elif schema['name'] == 'related_files':
+            entity_schema = schema['type']['items']
+        else:
+            assert False, schema
         for field in entity_schema['fields']:
             if field['name'] not in object_:
-                if name == 'files':
-                    default_value = None
+                if name == 'files' or name == 'related_files':
+                    # FIXME: Change 'string' to 'null' in https://github.com/DataBiosphere/azul/issues/2462
+                    if 'string' in field['type']:
```

Comment:
Why is this using `in` and not `==`?

Reply:
All `_nullable_to_pfb_types` are lists that contain `string`, but not all only contain `string`.